unified-scheduler-pool/src/lib.rs

```diff
@@ -0,0 +1,740 @@
+            batch,
+            transaction_indexes: vec![index],
+        };
+
+        *result = execute_batch(
+            &batch_with_indexes,
+            bank,
+            handler_context.transaction_status_sender.as_ref(),
+            handler_context.replay_vote_sender.as_ref(),
+            timings,
+            handler_context.log_messages_bytes_limit,
+            &handler_context.prioritization_fee_cache,
+        );
+    }
+}
+
+// Currently, simplest possible implementation (i.e. single-threaded)
+// this will be replaced with more proper implementation...
+// not usable at all, especially for mainnet-beta
+#[derive(Debug)]
+pub struct PooledScheduler<TH: TaskHandler> {
+    inner: PooledSchedulerInner<Self, TH>,
+    context: SchedulingContext,
+    result_with_timings: Mutex<ResultWithTimings>,
+}
+
+#[derive(Debug)]
+pub struct PooledSchedulerInner<S: SpawnableScheduler<TH>, TH: TaskHandler> {
+    id: SchedulerId,
+    pool: Arc<SchedulerPool<S, TH>>,
+}
+
+impl<TH: TaskHandler> PooledScheduler<TH> {
+    fn do_spawn(pool: Arc<SchedulerPool<Self, TH>>, initial_context: SchedulingContext) -> Self {
+        Self::from_inner(
+            PooledSchedulerInner::<Self, TH> {
+                id: pool.new_scheduler_id(),
+                pool,
+            },
+            initial_context,
+        )
+    }
+}
+
+pub trait SpawnableScheduler<TH: TaskHandler>: InstalledScheduler {
+    type Inner: Debug + Send + Sync;
+
+    fn into_inner(self) -> (ResultWithTimings, Self::Inner);
+
+    fn from_inner(inner: Self::Inner, context: SchedulingContext) -> Self;
+
+    fn spawn(pool: Arc<SchedulerPool<Self, TH>>, initial_context: SchedulingContext) -> Self
+    where
+        Self: Sized;
+}
+
+impl<TH: TaskHandler> SpawnableScheduler<TH> for PooledScheduler<TH> {
+    type Inner = PooledSchedulerInner<Self, TH>;
+
+    fn into_inner(self) -> (ResultWithTimings, Self::Inner) {
+        (
+            self.result_with_timings.into_inner().expect("not poisoned"),
+            self.inner,
+        )
+    }
+
+    fn from_inner(inner: Self::Inner, context: SchedulingContext) -> Self {
+        Self {
+            inner,
+            context,
+            result_with_timings: Mutex::new((Ok(()), ExecuteTimings::default())),
+        }
+    }
+
+    fn spawn(pool: Arc<SchedulerPool<Self, TH>>, initial_context: SchedulingContext) -> Self {
+        Self::do_spawn(pool, initial_context)
+    }
+}
+
+impl<TH: TaskHandler> InstalledScheduler for PooledScheduler<TH> {
+    fn id(&self) -> SchedulerId {
+        self.inner.id
+    }
+
+    fn context(&self) -> &SchedulingContext {
+        &self.context
+    }
+
+    fn schedule_execution(&self, &(transaction, index): &(&SanitizedTransaction, usize)) {
+        let (result, timings) = &mut *self.result_with_timings.lock().expect("not poisoned");
+        if result.is_err() {
+            // just bail out early to short-circuit the processing altogether
```

Comment:
btw, I'll create a follow-up pr to `::schedule_execution()` return a result to mark block as dead as early as possible... let this pr ship for now, please. lol